docs/weaver_code_anatomy.md

```diff
@@ -112,63 +123,70 @@ sections:
 ## Interview order
 Controls the order in which your screens are shown.
 
-There is some AssemblyLine code that comes before your own custom interview order code. You will probably leave this code alone:
-
-1. `nav.set_section()` comes after `al_intro_screen` and `a_258e_motion_for_impoundment_intro` so that the user can't click to edit their answers before they've actually been asked any questions.
-1. `allowed_courts` is currently connected to MassAccess, but it will be expanded in the future. It allows the developer to limit which courts the person filling out the form can pick from. That makes it easier for them to pick the right court.
-1. `user_role` tells AssemblyLine which questions to ask about the main party listed on the form.
-
-Code for your pages comes next. All your questions should be triggered in here. You will probably make major edits to the code here, changing the order and adding branching logic.
-
-<!-- TODO: Why is `allowed_courts` set lower down as opposed to before all questions or in some separate block? -->
-<!-- TODO: Should `nav.set_section()` come after the first actual question is asked if that's really the purpose? -->
+The full interview order block will look something like this. We'll go over each line individually below.
 
 ```yml
 id: interview_order_a_258e_motion_for_impoundment
 code: |
-  # This is a placeholder to control logic flow in this interview
-  # Intro screen/splash screen
-  al_intro_screen
-  a_258e_motion_for_impoundment_intro
   # Set the allowed courts for this interview
   allowed_courts = interview_metadata["a_258e_motion_for_impoundment"]["allowed courts"]
   nav.set_section('review_a_258e_motion_for_impoundment')
   user_role = 'plaintiff'
   one_of_your_custom_questions
+  users[0].phone_number
   another_of_your_custom_questions
   # Set the answer file name.
   set_parts(subtitle=str(users))
   set_progress(16.67)
+  trial_court.division
+  interview_order_a_258e_motion_for_impoundment = True
+---
+```
+
+1. `allowed_courts` allows the developer to limit which courts the person filling out the form can pick from, making it easier for them to pick the right court. By default, it's using the same values that are in the [metadata block](#assemblyline-metadata).
+1. `nav.set_section()` comes after `al_intro_screen` and `a_258e_motion_for_impoundment_intro` so that the user can't click to edit their answers before they've actually been asked any questions.
+1. `user_role` tells AssemblyLine which questions to ask about the main party listed on the form.
```

Comment:
I'm curious - what relation does `user_role` have to `typical role`?

Reply:
I think `user_role` is the runtime variable, and `typical role` is the "compile time" value. If [typical role anything but 'unknown'](https://github.com/SuffolkLITLab/docassemble-AssemblyLine/blob/48705945f9336adce66c62140bdc3cfa9f8c0ac1/docassemble/AssemblyLine/data/questions/al_code.yml#L175), then user_role is just set to that, and if it is unknown, then we ask for the `user_role`. I only assume we set it in the weaver because it's clearer what's happening, without a bunch of DA magic happening behind the scenes, and maybe for backwards compatibility?

Reply:
#134